src/realm/table.cpp

```diff
@@ -1053,12 +1053,20 @@ bool Table::set_embedded(bool embedded)
         });
 
         if (has_backlink_columns) {
-            for (auto o : *this) {
-                // each object should be owned by one and only one parent
-                if (o.get_backlink_count() != 1) {
+            for (auto object : *this) {
+                if (object.get_backlink_count() == 0) {
+                    // If the object does not have any backlinks it can be deleted.
+                    object.remove();
```

Comment:
Not sure it is a good idea to remove objects if the whole table does not fulfill the criteria of being embedded.